### DISCUSSION TITLE
Pivot Table Freeze Columns or Rows

### DISCUSSION THREAD

**Discussion by LeoDiep** [apache/superset#23404] [Ideas]
I would like the ability to freeze a row or column from moving when I scroll down or over on the pivot table chart.

If you have a large pivot table that involves scrolling in either direction you can easily lose track of which row or column you are looking at.

By freezing the column or row on scrolling we do not have that issue. I see this as a very useful feature for large tables.

Please vote for this to make it happen. Thank you guys

**Comment by LeoDiep** [comment]
Hi all, 
is there any update on this simple but important feature?

**Comment by LeoDiep** [comment]
Up

**Comment by jamzl** [comment]
This would be a really useful feature for my team.

**Comment by kkeveryday** [comment]
+, vote for adding this feature

**Reply by kkeveryday** [reply]
ayo guys, find some css solution (example for first column):
`.css-h8dzev {
  overflow: auto;
}`
`.pvtTable th {
  position: sticky;
  left: 0;
  z-index: 2;
}`

and check this: https://www.youtube.com/watch?v=YEzGe7uY2Oo

**Reply by ChristopherKi** [reply]
In case of multiple pivot table columns, you can also specify, which column to freeze by specifying the `colspan` of the `th` element. If you also have different charts, you can additionally specify the `chart-id` to use different settings for different charts

Full Example:
```

#chart-id-182 {
  .pvtTable th[colspan="1"] { position: sticky; left: 0; z-index: 2; }
}
```

**Reply by LeoDiep** [reply]
Hi there, would you guys please elaborate these answers more, how to do it on Superset frontend, I am not a dev so I couldn't follow it. Sorry

**Reply by ChristopherKi** [reply]
@LeoDiep you can configure this by going to the dashboard -> edit dashboard -> 3-dot-menu edit CSS -> Live CSS editor

**Comment by alexmattos** [comment]
+1

**Reply by dmshem** [reply]
+1
Tables are very useful view type

**Comment by dmshem** [comment]
This code work on a table view (not the pivot table):
```
div.superset-chart-table table td:first-child,
div.superset-chart-table table th:first-child
{
  position: sticky; left: 0; z-index: 2;
  background: #fff !important;
}
```

**Comment by bluespring2812** [comment]
+, vote for adding this feature
